packages/components/button/src/Button/Button.tsx

```diff
@@ -73,6 +73,10 @@ function _Button<E extends React.ElementType = typeof BUTTON_DEFAULT_TAG>(
         >
           {React.cloneElement(icon, {
             size: icon.props.size ?? `${size === 'large' ? 'medium' : 'small'}`,
+            // We need to pass the color to the icons to enable the usaged of the V5 icons
+            // it may change in the future
+            color:
+              (variant === 'transparent' && icon.props.color) || 'currentColor',
```

Comment:
```suggestion
              variant === 'transparent' && icon.props.variant === undefined ? icon.props.color : 'currentColor',
```

Would this work?

Reply:
what is interesting to me is, that this issue only appears in user_interface not when i try to use any of the new icons in storybook or anywhere else. (which is why i can not tell you if either of these variations are working)

Reply:
@Lelith what do you mean by "this issue", exactly? What's not working? 🤔

Reply:
the wrong color of the icon. on storybook the styles do not overwrite the color set from the button.

Reply:
<img width="665" alt="image" src="https://github.com/contentful/forma-36/assets/1789174/ef57323b-94b3-4cfa-9e6b-373ba706fa5b">
this class is not added in storybook examples
<img width="615" alt="image" src="https://github.com/contentful/forma-36/assets/1789174/26e521e4-ed7c-4aa9-a62a-78df01a7a000">

Reply:
@Lelith This is because in Storybook the IconButton isn't using the V5 Icons, and still uses the V4. But since Phosphor icons works with the color prop, we need to pass it down, which causes the `Box` component that is used by the `Icon` to add it to the CSS

Reply:
@massao as i was trying to replicate the behaviour, i actually installed them in my local working branch and my example is using a v5 icon.

Reply:
@denkristoffer We would need something like this, so we only keep the icon.props.color if it exists, otherwiser it would be reset to the default values on the V5 icons 🤔 

```suggestion
              (variant === 'transparent' && icon.props.variant === undefined && icon.props.color) || 'currentColor',
```

Reply:
Anyways, it's just an observation, i think the fix should still work and cover our bases. Let's merge and check it out in the user_interface branches